pkg/plugins/runtime/gateway/match/policy.go

```diff
@@ -60,3 +61,58 @@ func OldestPolicy(policies []model.Resource) model.Resource {
 
 	return sorted[0]
 }
+
+// BestPolicyForDestination returns the retry policy for the collection
+// of forwarding target. This is conceptually a bit subtle because a
+// forwarding target can have multiple destinations, each of which is a
+// distinct service.  However, there are some relatively obvious rules that
+// we can use to determine policy.
+//
+// 1. If all the destinations are the same service, use that policy.
+// 2. If there are multiple destinations, prefer a wildcard policy.
+// 3. Everything else being equal, older policies are preferred.
+func BestPolicyForDestination(
```

Comment:
should this be `BestConnectionPolicyForDestination`?

Reply:
Fixed.